lightning/src/ln/channel.rs

```diff
@@ -3589,10 +3603,22 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			return Err(ChannelError::Ignore("Cannot send value that would put us over the max HTLC value in flight our peer will accept"));
 		}
 
-		// Add additional reserve that avoids stuck channels in the case of fee spikes.
+		if !self.channel_outbound {
+			// Check that we won't violate their channel reserve by adding this HTLC.
+
+			let their_balance = self.channel_value_satoshis * 1000 - self.value_to_self_msat;
+			let chan_reserve_we_require_msat = Channel::<ChanSigner>::get_our_channel_reserve_satoshis(self.channel_value_satoshis);
+			// +1 for this HTLC, +2 for their fee spike buffer
+			// TODO: is the reason this added HTLC's amount can't count toward's the receiver's
+			// fee spike buffer because that doesn't work with existing HTLC output spend scripts?
+			let remote_commit_tx_fee_msat = self.commit_tx_fee_msat(self.htlc_count_next_remote_commit_tx() + 1 + 2);
```

Comment:
Same here you if you enforce a 2*HTLC feerate as fee spike buffer as HTLC sending, remote initiator can't receive anymore incoming balance to receive payment.

It's not what the spec mandates?
" if it is not responsible for paying the Bitcoin fee:
SHOULD NOT offer amount_msat if, once the remote node adds that HTLC to its commitment transaction, it cannot pay the fee for the updated local or remote transaction at the current feerate_per_kw while maintaining its channel reserve."